girder_volview/__init__.py

```diff
@@ -49,6 +50,47 @@
 LARGE_IMAGE_CONFIG_FOLDER = "large_image.config_folder"
 
 
+def hasLoadableFile(files):
+    for fileEntry in files:
+        if isLoadableFile(fileEntry[1]):
+            return True
+    return False
+
+
+@access.public(cookie=True, scope=TokenScope.DATA_READ)
+@boundHandler
+@autoDescribeRoute(
+    Description(
+        "Check if item has files VolView can load.  If so, return {loadable:true}."
+    )
+    .modelParam("itemId", model=Item, level=AccessType.READ)
+    .produces(["application/json"])
+    .errorResponse("ID was invalid.")
+    .errorResponse("Read access was denied for the folder.", 403)
+)
+def volViewLoadableItem(self, item):
+    files = Item().fileList(item, subpath=False, data=False)
+    loadable = hasLoadableFile(files)
+    return {"loadable": loadable}
+
+
+@access.public(cookie=True, scope=TokenScope.DATA_READ)
+@boundHandler
+@autoDescribeRoute(
+    Description(
+        "Check if folder has files VolView can load.  If so, return {loadable:true}."
+    )
+    .modelParam("folderId", model=Folder, level=AccessType.READ)
+    .produces(["application/json"])
+    .errorResponse("ID was invalid.")
+    .errorResponse("Read access was denied for the folder.", 403)
+)
+def volViewLoadableFolder(self, folder):
+    files = Folder().fileList(folder, subpath=False, data=False)
```

Comment:
Is this using the appropriate database aggregation?

Reply:
Sadly it isn't, but that should be changed in Girder.  This is a perfectly good approach.